### ISSUE TITLE
MD049/50: Incorrect range/fixinfo if identical errors on the same line

### ISSUE THREAD

**Issue by Billiam** [DavidAnson/markdownlint#486]
Example: 

```markdown
**a** __b__ __b__
```

The above will generate two errors for inconsistent strong style for the two `__b__` elements, but both errors will share the same range and fixinfo, pointing to only the first instance.

Obviously markdown-it doesn't provide this information with the token, and that range info comes from `helpers.getRangeAndFixInfoIfFound`, which just uses `indexOf` against the markdown line. This works fine as long as the elements don't have identical content like above.

Instead, I think `getRangeAndFixInfoIfFound` could accept an offset or instance number instead of only returning info for the first result.

I assume this affects other inline checks but I'm not sure which offhand.

**Comment by DavidAnson** [comment]
Agree, but I expect this is very unlikely in practice. And if it does come up, the next iteration will fix that second violation.
